tests/utilities/test_data.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+import torch
+from packaging.version import parse
+from torch.utils.data.dataloader import DataLoader
+from torch.utils.data.dataset import IterableDataset
+
+from pytorch_lightning import Trainer
+from pytorch_lightning.utilities.data import has_iterable_dataset, has_len
+from tests.base import EvalModelTemplate
+
+
+@pytest.mark.xfail(
+    parse(torch.__version__) < parse("1.4.0"),
+    reason="IterableDataset with __len__ before 1.4 raises",
+)
+def test_warning_with_iterable_dataset_and_len(tmpdir):
+    """ Tests that a warning messages is shown when an IterableDataset defines `__len__`. """
```

Comment:
```suggestion
    """ Tests that a warning message is shown when an IterableDataset defines `__len__`. """
```
nice to see this test for the warning 👍

Reply:
The author is not me. But I learned it's a good practice to add tests for the warnings. Thank you.